src/flextopo.jl

```diff
@@ -0,0 +1,795 @@
+        #make sure ds does not exceed 1 
+        pref_recharge = flextopo.qrootzone[i] .* min.(flextopo.ds[i], 1)
+        pref_recharge_sum_classes = sum(pref_recharge .* flextopo.hrufrac[i])
+
+        Qcap_sum_classes = sum(flextopo.qcapillary[i] .* flextopo.hrufrac[i])
+        Qperc_sum_classes = sum(flextopo.qpercolation[i] .* flextopo.hrufrac[i])
+
+        Qsin = pref_recharge_sum_classes + Qperc_sum_classes - Qcap_sum_classes
+        qslow = Qsin + flextopo.slowstorage[i] # if at start store is not empty 
+        slowstorage = 0.0
+
+        qfast_tot =
+            sum(flextopo.qfast[i] .* flextopo.hrufrac[i]) +
+            sum(flextopo.qhortonrun[i] .* flextopo.hrufrac[i])
+        runoff = max(0, qslow + qfast_tot)
+
+        wb_slow = Qsin - qslow - slowstorage + flextopo.slowstorage[i]
+
+        #update
+        flextopo.qfast_tot[i] = qfast_tot
+        flextopo.states_m[i] =
+            flextopo.states_m[i] +
+            sum(flextopo.states_[i] .* flextopo.hrufrac[i]) +
+            flextopo.slowstorage[i]
+        flextopo.runoff[i] = runoff
+        flextopo.qrootzoneslow_m[i] = pref_recharge_sum_classes
+        flextopo.qpercolation_m[i] = Qperc_sum_classes
+        flextopo.qcapillary_m[i] = Qcap_sum_classes
+        flextopo.qslow[i] = qslow
+        flextopo.slowstorage[i] = slowstorage
+        flextopo.wb_slow[i] = wb_slow
+    end
+end
+
+function common_slow_storage(flextopo::FLEXTOPO, config)
+    for i = 1:flextopo.n
+        #make sure ds does not exceed 1 
+        pref_recharge = flextopo.qrootzone[i] .* min.(flextopo.ds[i], 1)
+        pref_recharge_sum_classes = sum(pref_recharge .* flextopo.hrufrac[i])
+
+        Qcap_sum_classes = sum(flextopo.qcapillary[i] .* flextopo.hrufrac[i])
+        Qperc_sum_classes = sum(flextopo.qpercolation[i] .* flextopo.hrufrac[i])
+
+        Qsin = pref_recharge_sum_classes + Qperc_sum_classes - Qcap_sum_classes
+        slowstorage = flextopo.slowstorage[i] + Qsin
+
+        qslow = min(flextopo.slowstorage[i], flextopo.slowstorage[i] * flextopo.ks[i])
+        slowstorage = slowstorage - qslow
+
+        qfast_tot =
+            sum(flextopo.qfast[i] .* flextopo.hrufrac[i]) +
+            sum(flextopo.qhortonrun[i] .* flextopo.hrufrac[i])
+        runoff = max(0, qslow + qfast_tot)
+
+        wb_slow = Qsin - qslow - slowstorage + flextopo.slowstorage[i]
+
+        #update
+        flextopo.qfast_tot[i] = qfast_tot
+        flextopo.states_m[i] =
+            flextopo.states_m[i] +
+            sum(flextopo.states_[i] .* flextopo.hrufrac[i]) +
+            flextopo.slowstorage[i]
+        flextopo.runoff[i] = runoff
+        flextopo.qrootzoneslow_m[i] = pref_recharge_sum_classes
+        flextopo.qpercolation_m[i] = Qperc_sum_classes
+        flextopo.qcapillary_m[i] = Qcap_sum_classes
+        flextopo.qslow[i] = qslow
+        flextopo.slowstorage[i] = slowstorage
+        flextopo.wb_slow[i] = wb_slow
+    end
+end
+
+function watbal(flextopo::FLEXTOPO, config)
```

Comment:
In these and the functions above the config is not used. Should we remove it as an argument?

Reply:
My preference would be to remove the `config` argument if it is not used in a function.

Reply:
Yeah let's remove them. These are internal functions anyway, it's more clear if the arguments are actually used.

Reply:
I will remove them!